src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -282,6 +282,24 @@ class GcsActorManager : public rpc::ActorInfoHandler {
   /// change.
   void SchedulePendingActors();
 
+  /// Destroy an actor that has gone out of scope. This cleans up all local
+  /// state associated with the actor and marks the actor as dead. For owned
+  /// actors, this should be called when all actor handles have gone out of
+  /// scope or the owner has died.
+  /// NOTE: This method can be called multiple times in out-of-order and should be
+  /// idempotent.
+  ///
+  /// \param[in] actor_id The actor id to destroy.
+  /// \param[in] death_cause The reason why actor is destroyed.
+  void DestroyActor(const ActorID &actor_id, const rpc::ActorDeathCause &death_cause);
```

Comment:
In my option it would be better not involving rpc stuff in the gcs_actor_manager_test.cc test cases before, so core logical not rely not communication layer. I am not sure why RegisterActor method is made public , it seems that this method is used the way as I thought.  But yes I am totally ok with not making DestroyActor public, maybe instead using HandleKillActorViaGcs() in test case?

Reply:
We don't have tests that directly use Handle* functions I think. Why don't we just follow the existing pattern from other tests that test failures? I think they use public interface such as `OnNodeDead`.

Reply:
(and maybe FRIEND_TEST is a bad idea)

Reply:
Thanks for the advise, I would like align with other test cases by using OnNodeDead.